outport/outport_test.go

```diff
@@ -580,3 +581,90 @@ func TestOutport_SaveBlockDriverIsNotStuck(t *testing.T) {
 
 	assert.Equal(t, uint32(2), atomicGo.LoadUint32(&numLogDebugCalled))
 }
+
+func TestOutport_SettingsRequestAndReceive(t *testing.T) {
+	t.Parallel()
+
+	expectedErr := errors.New("expected error")
+	t.Run("RegisterHandlerForSettingsRequest errors, should not add the driver", func(t *testing.T) {
+		t.Parallel()
+
+		driver := &mock.DriverStub{
+			RegisterHandlerForSettingsRequestCalled: func(handlerFunction func()) error {
+				return expectedErr
+			},
+		}
+
+		outportHandler, _ := NewOutport(time.Second, outportcore.OutportConfig{})
+		err := outportHandler.SubscribeDriver(driver)
+		assert.Equal(t, expectedErr, err)
+		require.False(t, outportHandler.HasDrivers())
+	})
+	t.Run("SetCurrentSettings errors, should not panic", func(t *testing.T) {
+		t.Parallel()
+
+		defer func() {
+			r := recover()
+			if r != nil {
+				assert.Fail(t, fmt.Sprintf("should have not failed %v", r))
+			}
+		}()
+
+		currentSettingsCalled := false
+		var callback func()
+		driver := &mock.DriverStub{
+			RegisterHandlerForSettingsRequestCalled: func(handlerFunction func()) error {
+				callback = handlerFunction
+
+				return nil
+			},
+			SetCurrentSettingsCalled: func(config outportcore.OutportConfig) error {
+				currentSettingsCalled = true
+				return expectedErr
+			},
+		}
+
+		outportHandler, _ := NewOutport(time.Second, outportcore.OutportConfig{})
+		err := outportHandler.SubscribeDriver(driver)
+
+		callback()
+
+		assert.Nil(t, err)
+		assert.True(t, currentSettingsCalled)
```

Comment:
Could've checked before `callback()` that this is set to false

Reply:
done